### DISCUSSION TITLE
Modoboa Webmail doesn't work after publich IP change

### DISCUSSION THREAD

**Discussion by TheBigM85** [modoboa/modoboa#2004] [Help]
# Impacted versions

* Modoboa: 1.15.0 (latest Version)
* installer used: Yes
* Webserver: Nginx

# Steps to reproduce
Change the public IP to another one.

# Current behavior
All looks good. Webserver works (Websites and Webmail login page loads). Mail incoming an send is working.

# Expected behavior
After fill in Webmail forms it should load the Webmail, but get error 504 - Gateway Time-out

# Video/Screenshot link (optional)

Additional:
After changing my public IP (via openvpn) from portunity OpenVPN to own one (OpenVPN via vServer to my Server with modoboa, NextCloud and Websites at Home; is the OpenVPN Client an get the public IP via OpenVPN Server; works very well till changing the Server/IP) the public IP changed (Firewall rules and Forwarding all the same and works). But since changing the IP (dns in domains are corrected to new IP) the webmailer wont load. My additional Websites on server works and nextcloud is working too without problems. Mail sending and incoming Mails via modoboa Mailserver works too. Only Problem is Webmail and the Error 504 - Gateway Time-out.
Does modoboa save the public IP in any config and now has problems? Can anyone help my to fix this issue?

Thanks a lot
(sorry if english is not very good; is school english; i'm from germany :-) )

Greats

TheBigM85

**Comment by TheBigM85** [comment]
Additional Info:
This is the error from /var/log/nginx/mailserver.domain.de-error.log
2020/08/19 20:46:28 [error] 1219#1219: *132 connect() failed (111: Connection refused) while connecting to upstream, client: 84.131.XXX.XXX, server: mailserver.domain.de, request: "REPORT /radicale/user@domain.de/Mario/ HTTP/1.1", upstream: "http://[::1]:5232/user@domain.de/Mario/", host: "mailserver.domain.de"

The Webmail or Mailserver Dashboard is loading and i can add mailboxes and so on. But only Webmailer is not working and run in 504 - Time Out . It is not possible to login to any mailbox on webmailer. Outlook or Thunderbird is working without problems.

**Comment by tonioo** [comment]
@TheBigM85 Mododoa does not store any public IP. According to this log, there is an issue with Radicale...
Can you check uwsgi logs?

**Comment by TheBigM85** [comment]
One Point is, that my python was broken but now fixed it (i see no errors for python at now). I rebooted the Server, but can still not login to Webmail (Admin Dashboard, NextCloud and so on are ok).
Log for uwsgi modoboa-instance was loging this [/var/log/uwsgi/app/modoboa_instance.log] --> attached File.
###############################
[uwsgi_log_modoboa.txt](https://github.com/modoboa/modoboa/files/5119323/uwsgi_log_modoboa.txt)

Before i fixed broken python, i had this error get to root-Mail:
Traceback (most recent call last):
  File "/srv/modoboa/instance/manage.py", line 21, in <module>
    main()
  File "/srv/modoboa/instance/manage.py", line 17, in main
    execute_from_command_line(sys.argv)
  File "/srv/modoboa/env/lib/python3.6/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/srv/modoboa/env/lib/python3.6/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/srv/modoboa/env/lib/python3.6/site-packages/django/core/management/base.py", line 323, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/srv/modoboa/env/lib/python3.6/site-packages/django/core/management/base.py", line 364, in execute
    output = self.handle(*args, **options)
  File "/srv/modoboa/env/lib/python3.6/site-packages/modoboa_amavis/management/commands/qcleanup.py", line 59, in handle
    Msgs.objects.filter(time_num__lt=limit).delete()
  File "/srv/modoboa/env/lib/python3.6/site-packages/django/db/models/query.py", line 711, in delete
    deleted, _rows_count = collector.delete()
  File "/srv/modoboa/env/lib/python3.6/site-packages/django/db/models/deletion.py", line 266, in delete
    self.data[model] = sorted(instances, key=attrgetter("pk"))
TypeError: '<' not supported between instances of 'memoryview' and 'memoryview'

But i think, this should be fixed now, because python is now ok.
This Services run with python to and there are no service errors:
networkd-dispatcher, fail2ban-server

If i check Version and alternatives, i get following:
update-alternatives --list python
/usr/bin/python2.7
/usr/bin/python3.6
/usr/bin/python3.7

python --version
Python 3.6.9

Hope this helps, to get my problem solved.

Thx

**Comment by tonioo** [comment]
@TheBigM85 The trace you're showing is indeed a bug but it is not causing your login issue. What happens now when you try to login to the webmail? Still a 504 error?

**Comment by TheBigM85** [comment]
Yes. I still get 504 Gateway Timeout.
As Workaround i setup Roundcube and connect this to modoboa mailserver and config my vhosts to Roundcube. That works for the moment. but in the long run I would prefer modoboa webmail to work again.
What can it be?

Greats
TheBitM85

**Comment by tonioo** [comment]
@TheBigM85 Have you tried to enable debug mode? (set DEBUG to True inside settings.py and reload uwsgi) You might see more information in logs...

**Comment by TheBigM85** [comment]
Hi.
Thanks for your comment and hint. I'm do this and now get the following infos / errors in uwsgi-log, but i don't understand, where the Problem is. Can you tell or help me please.
This is the part from the log from restart the uwsgi service and my failed try to login:

########################################################################
Fri Sep 11 21:08:23 2020 - SIGINT/SIGQUIT received...killing workers...
Fri Sep 11 21:08:24 2020 - worker 1 buried after 1 seconds
Fri Sep 11 21:08:24 2020 - worker 2 buried after 1 seconds
Fri Sep 11 21:08:24 2020 - goodbye to uWSGI.
Fri Sep 11 21:08:24 2020 - VACUUM: unix socket /run/uwsgi/app/modoboa_instance/socket removed.
Fri Sep 11 21:08:25 2020 - *** Starting uWSGI 2.0.15-debian (64bit) on [Fri Sep 11 21:08:25 2020] ***
Fri Sep 11 21:08:25 2020 - compiled with version: 7.3.0 on 28 September 2018 15:41:15
Fri Sep 11 21:08:25 2020 - os: Linux-5.4.0-1023-azure #23~18.04.1-Ubuntu SMP Thu Aug 20 14:46:48 UTC 2020
Fri Sep 11 21:08:25 2020 - nodename: mail
Fri Sep 11 21:08:25 2020 - machine: x86_64
Fri Sep 11 21:08:25 2020 - clock source: unix
Fri Sep 11 21:08:25 2020 - pcre jit disabled
Fri Sep 11 21:08:25 2020 - detected number of CPU cores: 2
Fri Sep 11 21:08:25 2020 - current working directory: /
Fri Sep 11 21:08:25 2020 - writing pidfile to /run/uwsgi/app/modoboa_instance/pid
Fri Sep 11 21:08:25 2020 - detected binary path: /usr/bin/uwsgi-core
Fri Sep 11 21:08:25 2020 - dropping root privileges as early as possible
Fri Sep 11 21:08:25 2020 - setgid() to 1001
Fri Sep 11 21:08:25 2020 - setuid() to 1001
Fri Sep 11 21:08:25 2020 - chdir() to /srv/modoboa/instance
Fri Sep 11 21:08:25 2020 - your processes number limit is 20197
Fri Sep 11 21:08:25 2020 - your memory page size is 4096 bytes
Fri Sep 11 21:08:25 2020 - detected max file descriptor number: 1024
Fri Sep 11 21:08:25 2020 - VirtualHosting mode enabled.
Fri Sep 11 21:08:25 2020 - lock engine: pthread robust mutexes
Fri Sep 11 21:08:25 2020 - thunder lock: disabled (you can enable it with --thunder-lock)
Fri Sep 11 21:08:25 2020 - uwsgi socket 0 bound to UNIX address /run/uwsgi/app/modoboa_instance/socket fd 3
Fri Sep 11 21:08:25 2020 - dropping root privileges after socket binding
Fri Sep 11 21:08:25 2020 - Python version: 3.6.9 (default, Jul 17 2020, 12:50:27)  [GCC 8.4.0]
Fri Sep 11 21:08:25 2020 - PEP 405 virtualenv detected: /srv/modoboa/env
Fri Sep 11 21:08:25 2020 - Set PythonHome to /srv/modoboa/env
Fri Sep 11 21:08:26 2020 - *** Python threads support is disabled. You can enable it with --enable-threads ***
Fri Sep 11 21:08:26 2020 - Python main interpreter initialized at 0x56077d85f770
Fri Sep 11 21:08:26 2020 - dropping root privileges after plugin initialization
Fri Sep 11 21:08:26 2020 - your server socket listen backlog is limited to 100 connections
Fri Sep 11 21:08:26 2020 - your mercy for graceful operations on workers is 60 seconds
Fri Sep 11 21:08:26 2020 - mapped 218304 bytes (213 KB) for 2 cores
Fri Sep 11 21:08:26 2020 - *** Operational MODE: preforking ***
Fri Sep 11 21:08:31 2020 - WSGI app 0 (mountpoint='') ready in 5 seconds on interpreter 0x56077d85f770 pid: 4843
Fri Sep 11 21:08:31 2020 - dropping root privileges after application loading
Fri Sep 11 21:08:31 2020 - spawned uWSGI master process (pid: 4843)
Fri Sep 11 21:08:31 2020 - spawned uWSGI worker 1 (pid: 4885, cores: 1)
Fri Sep 11 21:08:31 2020 - spawned uWSGI worker 2 (pid: 4886, cores: 1)
Fri Sep 11 21:08:38 2020 - WSGI app 1 (mountpoint='mail.XXXXXXXXXXX.domain|') ready in 0 seconds on interpreter 0x56077d85f770 pid: 4885
mail.XXXXXXXXXXX.domain [pid: 4885|app: 1|req: 1/1] 12.123.123.12 () {46 vars in 767 bytes} [Fri Sep 11 21:08:38 2020] GET / => generated 0 bytes in 1420 msecs (HTTP/1.1 302) 6 headers in 198 bytes (1 switches on core 0)
mail.XXXXXXXXXXX.domain [pid: 4885|app: 1|req: 2/2] 12.123.123.12 () {46 vars in 810 bytes} [Fri Sep 11 21:08:40 2020] GET /accounts/login/?next=/ => generated 3099 bytes in 497 msecs (HTTP/1.1 200) 8 headers in 442 bytes (1 switches on core 0)
Fri Sep 11 21:08:40 2020 - WSGI app 1 (mountpoint='mail.XXXXXXXXXXX.domain|') ready in 0 seconds on interpreter 0x56077d85f770 pid: 4886
mail.XXXXXXXXXXX.domain [pid: 4886|app: 1|req: 1/3] 12.123.123.12 () {48 vars in 837 bytes} [Fri Sep 11 21:08:40 2020] GET /jsi18n/ => generated 5505 bytes in 1296 msecs (HTTP/1.1 200) 5 headers in 171 bytes (2 switches on core 0)
mail.XXXXXXXXXXX.domain [pid: 4886|app: 1|req: 2/4] 34.345.345.34 () {52 vars in 855 bytes} [Fri Sep 11 21:08:54 2020] HEAD / => generated 0 bytes in 324 msecs (HTTP/1.1 302) 6 headers in 198 bytes (2 switches on core 0)
mail.XXXXXXXXXXX.domain [pid: 4885|app: 1|req: 3/5] 12.123.123.12 () {56 vars in 1125 bytes} [Fri Sep 11 21:08:54 2020] POST /accounts/login/ => generated 0 bytes in 1650 msecs (HTTP/1.1 302) 10 headers in 552 bytes (1 switches on core 0)
mail.XXXXXXXXXXX.domain [pid: 4886|app: 1|req: 3/6] 34.345.345.34 () {52 vars in 898 bytes} [Fri Sep 11 21:08:55 2020] HEAD /accounts/login/?next=/ => generated 3085 bytes in 499 msecs (HTTP/1.1 200) 8 headers in 442 bytes (1 switches on core 0)
mail.XXXXXXXXXXX.domain [pid: 4885|app: 1|req: 4/7] 12.123.123.12 () {50 vars in 973 bytes} [Fri Sep 11 21:08:55 2020] GET / => generated 0 bytes in 225 msecs (HTTP/1.1 302) 6 headers in 184 bytes (1 switches on core 0)
mail.XXXXXXXXXXX.domain [pid: 4885|app: 1|req: 5/8] 12.123.123.12 () {46 vars in 827 bytes} [Fri Sep 11 21:09:56 2020] GET /favicon.ico => generated 9447 bytes in 186 msecs (HTTP/1.1 404) 5 headers in 155 bytes (1 switches on core 0)
Fri Sep 11 21:11:07 2020 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) on request /webmail/ (ip 12.123.123.12) !!!
Fri Sep 11 21:11:07 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 296] during GET /webmail/ (12.123.123.12)
OSError: write error
mail.XXXXXXXXXXX.domain [pid: 4886|app: 1|req: 4/9] 12.123.123.12 () {50 vars in 989 bytes} [Fri Sep 11 21:08:56 2020] GET /webmail/ => generated 0 bytes in 131717 msecs (HTTP/1.1 200) 6 headers in 0 bytes (0 switches on core 0)
########################################################################################
Hope you or any other could help me to solve the problem.

thanks in advance.

TheBigM85

**Comment by TheBigM85** [comment]
If tried with "uwsgi_ignore_client_abort on;" in nginx vhost, but no luck.

Still get 504 Gateway Time-out

**Comment by TheBigM85** [comment]
Ok. Another additional Info:
I Rebooted the Server and tried again:
This Error:
1. SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) on request /webmail/ (ip 12.123.123.12) !!!
2. uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 296] during GET /webmail/ (12.123.123.12)
OSError: write error

only happens, when i close or disconnect to the 504 error Page. If i stay on Error page, the last log entry is
"mail.XXXXXXXXXXX.domain [pid: 4885|app: 1|req: 5/8] 12.123.123.12 () {46 vars in 827 bytes} [Fri Sep 11 21:09:56 2020] GET /favicon.ico => generated 9447 bytes in 186 msecs (HTTP/1.1 404) 5 headers in 155 bytes (1 switches on core 0)"

Next entry after the errors (after closing the wrbbrowser or leaving the page) are or is:
"mail.XXXXXXXXXXX.domain [pid: 4886|app: 1|req: 4/9] 12.123.123.12 () {50 vars in 989 bytes} [Fri Sep 11 21:08:56 2020] GET /webmail/ => generated 0 bytes in 131717 msecs (HTTP/1.1 200) 6 headers in 0 bytes (0 switches on core 0)"

(Time Stamps here are not correct; are copied from the log before, because absolutly the same errors).

Modoboa Admin-Backend / Admin-Page are still working.

**Comment by tonioo** [comment]
And do you see anything related in the mail.log file? Maybe dovecot 
errors or warnings?

Le 11/09/2020 à 21:54, TheBigM85 a écrit :

**Comment by TheBigM85** [comment]
No.
Mailserver is working fine. I can look if you want, but i can receive and send Mails without problems (all is working from Smarthphone, Outlook, Thunderbird and Roundcube as i writed is working too; only modoboa webmail wont work / load).

**Comment by TheBigM85** [comment]
I looked in mail.log, but there is nothing to see. All works fine, as i wrote before.

**Comment by TheBigM85** [comment]
Here is an example for dovecot from the mail.log:

Sep 11 22:02:08 mail dovecot: lmtp(XXXX@xxxxxxxxxxxx.domain): msgid=<modoboa/modoboa/issues/2004/691287308@github.com>: saved mail to INBOX

and for postfix:

Sep 11 22:02:08 mail postfix/smtpd[6555]: disconnect from out-25.smtp.github.com[192.30.252.208] ehlo=2 starttls=1 mail=1 rcpt=1 data=1 quit=1 commands=7

**Comment by TheBigM85** [comment]
Info for you @tonioo by the way....
Your Site "http://tonio.ngyn.org/" has Error "502 Bad Gateway" ;-)

**Comment by tonioo** [comment]
@TheBigM85 yeah I know, that's an old site I don't maintain anymore ;)
Maybe I could take a look at your server?

**Comment by TheBigM85** [comment]
@tonioo Hm. How would you like to do this "look at my Server". Where do you have to look?
I don't want to give root to anyone (i hope you understand this?!). Could we do this via TeamViewer or so? But when? I'm from germany / europe.

**Comment by tonioo** [comment]
@TheBigM85 One thing: have you tried to disable external queries made by the dashboard?

DISABLE_DASHBOARD_EXTERNAL_QUERIES = True

**Comment by TheBigM85** [comment]
> @TheBigM85 One thing: have you tried to disable external queries made by the dashboard?
> 
> DISABLE_DASHBOARD_EXTERNAL_QUERIES = True

Where does i have to add this line???

**Comment by TheBigM85** [comment]
Ok. I found this in settings.py and changed the value. After this i rebooted the server for changes to take affect but it doesn't help. Same as before. And if i let it load till the error will be showed, it takes about 10 Minutes that the site for Modoboa will be reachable again or i have to restart some services or restart the server or wait about 3 minutes an use another browser (its not anytime the same). What does this value? Can i change it back to false or is this no problem?

**Comment by TheBigM85** [comment]
@tonioo 
On Reboot i get this error in /var/mail/root:
#######################################
From root@mail.domain.com  Sat Sep 19 00:00:47 2020
Return-Path: <root@mail.domain.com> [<----|Domain is crypted from me (no error); there stands the right Domain.]
X-Original-To: root
Delivered-To: root@mail.domain.com
Received: by mail.domain.com (Postfix, from userid 0)
        id EE5EB1C00075; Sat, 19 Sep 2020 00:00:46 +0200 (CEST)
From: root@mail.domain.com (Cron Daemon)
To: root@mail.domain.com
Subject: Cron <root@mail>    $PYTHON $INSTANCE/manage.py qcleanup
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
X-Cron-Env: <PYTHON=/srv/modoboa/env/bin/python>
X-Cron-Env: <INSTANCE=/srv/modoboa/instance>
X-Cron-Env: <SHELL=/bin/sh>
X-Cron-Env: <HOME=/root>
X-Cron-Env: <PATH=/usr/bin:/bin>
X-Cron-Env: <LOGNAME=root>
Message-Id: <20200918220046.EE5EB1C00075@mail.domain.com>
Date: Sat, 19 Sep 2020 00:00:46 +0200 (CEST)

Traceback (most recent call last):
  File "/srv/modoboa/instance/manage.py", line 21, in <module>
    main()
  File "/srv/modoboa/instance/manage.py", line 17, in main
    execute_from_command_line(sys.argv)
  File "/srv/modoboa/env/lib/python3.6/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/srv/modoboa/env/lib/python3.6/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/srv/modoboa/env/lib/python3.6/site-packages/django/core/management/base.py", line 323, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/srv/modoboa/env/lib/python3.6/site-packages/django/core/management/base.py", line 364, in execute
    output = self.handle(*args, **options)
  File "/srv/modoboa/env/lib/python3.6/site-packages/modoboa_amavis/management/commands/qcleanup.py", line 59, in handle
    Msgs.objects.filter(time_num__lt=limit).delete()
  File "/srv/modoboa/env/lib/python3.6/site-packages/django/db/models/query.py", line 711, in delete
    deleted, _rows_count = collector.delete()
  File "/srv/modoboa/env/lib/python3.6/site-packages/django/db/models/deletion.py", line 266, in delete
    self.data[model] = sorted(instances, key=attrgetter("pk"))
TypeError: '<' not supported between instances of 'memoryview' and 'memoryview'
###############################################
Maybe this is the reason for problems in Webmail?????

**Comment by TheBigM85** [comment]
Hi.

Is there now any other hint for my problem, to solve this?

**Comment by tonioo** [comment]
@TheBigM85 The setting you changed just enable or disable HTTP requests made by the dashboard to display the News (RSS feed) anv the projects opened to sponsoring. If you set the value to True, it disables those queries which can lead to timeout errors sometimes. That's why I asked you that but looks like the problem is elsewhere...
The error you mentioned is related to the amavis plugin and is caused by a cron task so it has nothing to do with the webmail. I really don't know what could be the problem on your server... 
Have you tried to manually load another page, like the admin part? Does your error only happen with the webmail?

**Comment by TheBigM85** [comment]
> @TheBigM85 The setting you changed just enable or disable HTTP requests made by the dashboard to display the News (RSS feed) anv the projects opened to sponsoring. If you set the value to True, it disables those queries which can lead to timeout errors sometimes. That's why I asked you that but looks like the problem is elsewhere...
> 
> The error you mentioned is related to the amavis plugin and is caused by a cron task so it has nothing to do with the webmail. I really don't know what could be the problem on your server... 
> 
> Have you tried to manually load another page, like the admin part? Does your error only happen with the webmail?

Hi.
Thanks for your answer / reply.
As i many times wrote before, all other is working (like admin page, new installed roundcube and all my other pages (include nextcloud)). Only Webmail doesn't work. Modoboa Admin-Page is working.

**Comment by TheBigM85** [comment]
Today i checked / tested again.
Now there is no 504, but there is now the following error:
Fehler: [b'[AUTHENTICATIONFAILED] Authentication failed.']

But the credentials are ok / right. I checked this with 3 of my adresses and copy and paste them from my KeePass -> same error.

**Comment by tonioo** [comment]
@TheBigM85 Could you enable dovecot auth debug mode and check your mail.log file?

**Comment by TheBigM85** [comment]
Hello.

Sorry for late answer. I do activate the debug-mode, but there is nothing in mail.log for the Mail and time, where i testetd from. 
Mail.log looks ok. Only in nginx error-log for the domain, there is an entry for timeout like this: 
2020/10/11 23:28:01 [error] 39599#39599: *50033 upstream timed out (110: Connection timed out) while reading response header from upstream, client: 93.XXX.XXX.XXXx, server: mail.domain-xxxx.com, request: "GET /webmail/ HTTP/1.1", upstream: "uwsgi://unix:/run/uwsgi/app/modoboa_instance/socket", host: "mail.domain-xxxx.com", referrer: "https://mail.domain-xxxx.com/accounts/login/?next=/"

All other is working well like before. Actualy i get error "504 Gateway Time-out", when i try to login to modoboa webmail. Modoboa Admin Interface for Domains an so on is working well.
The last error i posted ([b'[AUTHENTICATIONFAILED] Authentication failed.']) does not appear again now.

**Comment by tonioo** [comment]
@TheBigM85 And still nothing in uwsgi logs? Have you tried to play with the harakiri timeout value?

**Comment by sergioeix** [comment]
Hi @tonioo @TheBigM85 I know this is an old thread but i am facing an issue like this, when i try to login i get a conection timeout fail, after reboot same result, after restar uwsgi restart all works fine, but i want to share the log generated by uwsgi, hope this will help to fix the issue or find where the issue is.

Wed Oct 13 10:09:22 2021 - WSGI app 1 (mountpoint='xxx.xxxxxx.xx|') ready in 0 seconds on interpreter 0x55d8ef6f8520 pid: 1342
xxx.xxxxxx.xx [pid: 1342|app: 1|req: 1/1] 192.168.1.1 () {60 vars in 1189 bytes} [Wed Oct 13 10:09:22 2021] GET / => generated 0 bytes in 204 msecs (HTTP/1.1 302) 6 head$
Wed Oct 13 10:09:39 2021 - WSGI app 1 (mountpoint='xxx.xxxxxx.xx|') ready in 0 seconds on interpreter 0x55d8ef6f8520 pid: 1341
xxx.xxxxxx.xx [pid: 1341|app: 1|req: 1/2] 192.168.1.1 () {58 vars in 1046 bytes} [Wed Oct 13 10:09:39 2021] GET / => generated 0 bytes in 185 msecs (HTTP/1.1 302) 6 head$
xxx.xxxxxx.xx [pid: 1341|app: 1|req: 2/3] 192.168.1.1 () {58 vars in 1089 bytes} [Wed Oct 13 10:09:40 2021] GET /accounts/login/?next=/ => generated 3111 bytes in 67 mse$
xxx.xxxxxx.xx [pid: 1341|app: 1|req: 3/4] 192.168.1.1 () {58 vars in 1023 bytes} [Wed Oct 13 10:09:40 2021] GET /jsi18n/ => generated 5828 bytes in 35 msecs (HTTP/1.1 20$
xxx.xxxxxx.xx [pid: 1341|app: 1|req: 4/5] 192.168.1.1 () {70 vars in 1419 bytes} [Wed Oct 13 10:09:46 2021] POST /accounts/login/ => generated 0 bytes in 149 msecs (HTTP$
xxx.xxxxxx.xx [pid: 1341|app: 1|req: 5/6] 192.168.1.1 () {64 vars in 1279 bytes} [Wed Oct 13 10:09:46 2021] GET / => generated 0 bytes in 26 msecs (HTTP/1.1 302) 6 heade$
Wed Oct 13 10:13:44 2021 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) on request /dashboard/ (ip 192.168.1.1) !!!
Wed Oct 13 10:13:44 2021 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 296] during GET /dashboard/ (192.168.1.1)
OSError: write error
xxx.xxxxxx.xx [pid: 1342|app: 1|req: 2/7] 192.168.1.1 () {60 vars in 1209 bytes} [Wed Oct 13 10:09:23 2021] GET /dashboard/ => generated 0 bytes in 260983 msecs (HTTP/1.$
Wed Oct 13 10:13:44 2021 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) on request /favicon.ico (ip 192.168.1.1) !!!
Wed Oct 13 10:13:44 2021 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 296] during GET /favicon.ico (192.168.1.1)
OSError: write error
xxx.xxxxxx.xx [pid: 1342|app: 1|req: 3/8] 192.168.1.1 () {58 vars in 1132 bytes} [Wed Oct 13 10:13:44 2021] GET /favicon.ico => generated 0 bytes in 47 msecs (HTTP/1.1 4$
Wed Oct 13 10:13:44 2021 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) on request /favicon.ico (ip 192.168.1.1) !!!
Wed Oct 13 10:13:44 2021 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 296] during GET /favicon.ico (192.168.1.1)
OSError: write error
xxx.xxxxxx.xx [pid: 1342|app: 1|req: 4/9] 192.168.1.1 () {58 vars in 1123 bytes} [Wed Oct 13 10:13:44 2021] GET /favicon.ico => generated 0 bytes in 45 msecs (HTTP/1.1 4$
Wed Oct 13 10:14:08 2021 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) on request /dashboard/ (ip 192.168.1.1) !!!
Wed Oct 13 10:14:08 2021 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 296] during GET /dashboard/ (192.168.1.1)
OSError: write error

**Reply by tonioo** [reply]
@sergioeix Try to set DISABLE_DASHBOARD_QUERIES to True inside your settings.py file.

**Reply by sergioeix** [reply]
Hi @tonioo ty for your suggestion, i dont have this key on my settings.py i should add it? can you tell me what does this config key?
Ty and B/R

**Reply by tonioo** [reply]
What is your modoboa version?

**Reply by sergioeix** [reply]
Modoboa | 1.17.0

**Reply by tonioo** [reply]
Then this variable should be present in your settings.py file. If not, just add it with a value of True and reload uwsgi.

**Comment by TheBigM85** [comment]
@tonioo 
Hi. I'm looking for "DISABLE_DASHBOARD_QUERIES" in my settings.py in "/srv/odoboa/instance/instance" but there is only an "entry" / variable like "DISABLE_DASHBOARD_EXTERNAL_QUERIES" an is set to "false". Musst i set this to "true".
I still have the same problem as originally. I am currently still using roundcube for webmail. The Modoboa web interface for domains and accounts is still loading, but the webmail error still persists. Modoboa is on Version 1.17.0. Could this entry fix the problem? Musst i set "DISABLE_DASHBOARD_EXTERNAL_QUERIES" to "true" or must i add "DISABLE_DASHBOARD_QUERIES" to "true" in settings.py or does this not help for my problem?

**Comment by TheBigM85** [comment]
Ok. I tested to set "DISABLE_DASHBOARD_EXTERNAL_QUERIES" to "True" --> get internal Server error after Reboot. Set it back and do "upgrade" via "run.py" (for reinstall) --> Server is working again, but get farther the error "504 Gateway Time-Out" if i try to login with Mailaccount on modoboa Login / webmail login. Roundcube Webmail works fine and Modoboa Webinterface for Admin works fine too.

**Reply by tonioo** [reply]
Then your issue is different and not related external queries. What does your username look like? Is it an email address or just a name?

**Reply by TheBigM85** [reply]
For admin it is user "admin". For Webmail / Mail-Accounts it is mail address and password.

**Reply by tonioo** [reply]
And you see this webmail issue with all accounts?

**Reply by TheBigM85** [reply]
Yes. this problem is with all mail accounts

**Reply by tonioo** [reply]
Can you check the configuration of the webmail in the administration panel? (Modoboa > Parameters > Webmail)

**Reply by TheBigM85** [reply]
OK. That was the problem. I wrote back then that the public IP has changed due to the server move. The old IP address was still in the webmail configuration. Here I have now entered the FQDN and it works perfectly again. Thanks very much. So my problem is solved.
